prow/cluster/bazel-cache-deployment.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: bazel-cache
+  namespace: ${NAMESPACE}
+spec:
+  selector:
+    matchLabels:
+      app: bazel-cache
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: bazel-cache
+    spec:
+      nodeSelector:
+        kubernetes.io/hostname: prow-master-03 # This could be a proper label such as `role: bazel-cache`
```

Comment:
@dgn We need to decide where to deploy this pod.

Reply:
I'm hoping the master nodes are powerful enough. let's try with this one first